lightly/data/collate.py

```diff
@@ -36,6 +37,12 @@ class BaseCollateFunction(nn.Module):
     """
 
     def __init__(self, transform: torchvision.transforms.Compose):
+        warn(
```

Comment:
Can you add this function to the bottom of the file and then use it wherever you need it:
```python
def _deprecation_warning_collate_functions() -> None:
    warn(
        "Collate functions are deprecated and will be removed in favor of transforms in v1.4.0.\n"
        "Please use MultiViewCollate in `lightly.data.multi_view_collate` together with the correct transform for your model instead.\n"
        "See https://docs.lightly.ai/self-supervised-learning/examples/models.html for examples."
        category=DeprecationWarning,
    )
```

I'm not super happy with the wording. Maybe you can improve.

Note the change to `DeprecationWarning` for better visibility.